contracts/lib.rs

```diff
@@ -882,27 +945,15 @@ pub mod prosopo {
         #[ink(message)]
         #[ink(payable)]
         pub fn dapp_fund(&mut self, contract: AccountId) -> Result<(), Error> {
-            let transferred = self.env().transferred_value();
-            if self.dapps.get(contract).is_none() {
-                return err!(Error::DappDoesNotExist);
-            }
+            let mut dapp = self.get_dapp(contract)?;
 
-            let mut dapp = self
-                .dapps
-                .get(contract)
-                .ok_or_else(err_fn!(Error::DappDoesNotExist))?;
-            let total = dapp.balance + transferred;
-            dapp.balance = total;
-            if dapp.balance > 0 {
-                dapp.status = GovernanceStatus::Active;
-                self.env().emit_event(DappFund {
-                    contract,
-                    value: total,
-                });
-            } else {
-                // Suspended as dapp has no funds
-                dapp.status = GovernanceStatus::Suspended;
-            }
+            // check current contract for ownership
+            self.check_dapp_owner_is_caller(contract)?;
```

Comment:
Why would we restrict people from funding dapp accounts?

Reply:
As in why is there a caller/owner check? If so, I was on the fence about this. On one hand, restricting the owner to be the only one who can fund a dapp is clear and simple. On the other, allowing anyone to fund a dapp is flexible but lends itself to someone accidentally funding the wrong dapp.

So based on that last point I put in the owner/caller check, but more than happy to go the other way

Reply:
People can also accidentally send all their funds to the wrong address - we don't need to guard against this.

Reply:
okay cool, I will remove the guard against this